Convert height markers field to sequence flag

spacek531 opened the pull request:
Solves #25502, #25503

This PR does some minor touchups to Track Data.

Commit 1: add comments to explain how all the properties work to create a function track piece
Commit 2: convert the uint32_t field into a bitset
Commit 3: tweak the positions of height markers on flat ride bases
Commit 4: fix a bug where only some maze pieces had height markers

The height markers aren't on the same tiles as RCT2, but RCT2 doesn't do it in a logical way for most flat rides, so I tweaked them to feel more natural.

After tweaks:
<img width="783" height="506" alt="image" src="https://github.com/user-attachments/assets/2fec893d-d9c3-4479-9a00-75fdb6b001e5" />

OpenRCT2 before tweaks:
<img width="786" height="522" alt="image" src="https://github.com/user-attachments/assets/90538483-3575-4fef-aede-34e12cd8cb0f" />

RCT2 (screenshot courtesy of Marcel):

<img width="1056" height="541" alt="image" src="https://github.com/user-attachments/assets/b342d6b8-7a0f-4dbd-872d-8576bcb80812" />

MarcelVos96 commented:
I find the two diagonal markers on the motion sim to look a bit odd. Maybe the full four markers would look better? Either way it all worked fine for me.

MarcelVos96 commented:
What does that first image with the four height markers on an invisible ride represent? What ride is there?

spacek531 commented:
That is a 2x4 flat ride base of which no flat rides currently use, but is present in RCT2's code and continues to be present in OpenRCT2's.

spacek531 commented:
I've rebased the PR and refactored it to use the new FlagsHolder.

Commit 2 fully recreates the existing height marker behavior for every track piece.

Commit 3 enhances the height markers on flat rides:

<img width="1267" height="714" alt="image" src="https://github.com/user-attachments/assets/5dd51f3d-21cd-445f-86f5-ca598aa22704" />

Commit 4 tweaks the height marker placement on the new diagonal curves and banks to fix consistency and double-heightmarker issues:

Before:
<img width="1691" height="645" alt="image" src="https://github.com/user-attachments/assets/7e84d703-7243-433f-b27b-2a3f4c7257bd" />

After:
<img width="1718" height="593" alt="image" src="https://github.com/user-attachments/assets/6717a609-4485-4920-bde9-5e612034ac06" />

spacek531 commented:
I manually clang-formatted the file with `clang-format -i src/openrct2/ride/TrackData.cpp` and I guess that ignores clang-format on/off comments
